### PR TITLE
DDF-04342 Karaf 4.2.6 upgrade

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
Upgrades Karaf to 4.2.6
#### Select relevant component teams: 

@Bdthomson @Schachte 
#### Ask 2 committers to review/merge the PR and tag them here.

@brjeter
@vinamartin
#### How should this be tested?
Build / Install / Verify Functionality has not regressed

#### What are the relevant tickets?
Fixes: #https://github.com/codice/ddf/issues/4342

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

File: pom.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
     <parent>
         <groupId>ddf</groupId>
         <artifactId>ddf-parent</artifactId>
-        <version>1.0.5</version>
+        <version>1.0.6-SNAPSHOT</version>
```
Comment (bdeining) [inline pom.xml]: the maven deploy plugin was failing because of version mismatches
https://github.com/codice/ddf-parent/pull/11

File: catalog/admin/catalog-admin-poller-service/pom.xml [hunk 2]
```diff
@@ -87,6 +87,10 @@
                         <Embed-Dependency>
                             catalog-core-api-impl;scope=!test,platform-util
                         </Embed-Dependency>
+                        <Import-Package>
+                            !javax.activation,
+                            *
```
Comment (GabrielFabian) [inline catalog/admin/catalog-admin-poller-service/pom.xml]: nice 👍

Comment (bdeining) [reply catalog/admin/catalog-admin-poller-service/pom.xml]: 😕

File: pom.xml (hunk 1 above)
Comment (bdeining) [reply pom.xml]: might be best to leave this at snapshot for the time being, so we can test the deploy job and iterate before we release a new version of ddf-parent

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

File: features/kernel/src/main/feature/feature.xml [hunk 3]
```diff
@@ -85,6 +125,7 @@
     </feature>
 
     <feature name="guava" version="${project.version}">
+        <bundle>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.jsr305/${jsr305.version}</bundle>
```
Comment (Lambeaux) [inline features/kernel/src/main/feature/feature.xml]: ❓ Why was this needed for the upgrade? Are new dependencies looking for `javax.annotation` now?

**Comment by bdeining** [comment]
build now

File: features/kernel/src/main/feature/feature.xml (hunk 3 above)
Comment (bdeining) [reply features/kernel/src/main/feature/feature.xml]: Per the release notes, Karaf moved to Annotations 1.3 
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12311140&version=12344587
But it looks like Guava 20 is using a lower version of annotations, so we must explicitly start this bundle since it is no longer part of Karaf Standard / Enterprise profiles.

**Comment by Bdthomson** [comment]
image (11)

**Comment by bdeining** [comment]
build keeps failing on `Archiving artifacts`

**Comment by bdeining** [comment]
build now

**Comment by Schachte** [comment]
hero decent

**Comment by bdeining** [comment]
build now
